spyderlib/workers/updates.py

```diff
@@ -0,0 +1,173 @@
+    """
+    A QMessageBox derived widget that includes a QCheckBox aligned to the right
+    under the message and on top of the buttons.
+    """
+    def __init__(self, *args, **kwargs):
+        super(MessageCheckBox, self).__init__(*args, **kwargs)
+
+        self._checkbox = QCheckBox()
+
+        # Set layout to include checkbox
+        size = 9
+        check_layout = QVBoxLayout()
+        check_layout.addItem(QSpacerItem(size, size))
+        check_layout.addWidget(self._checkbox, 0, Qt.AlignRight)
+        check_layout.addItem(QSpacerItem(size, size))
+
+        # Access the Layout of the MessageBox to add the Checkbox
+        layout = self.layout()
+        layout.addLayout(check_layout, 1, 1)
+
+    # --- Public API
+    # Methods to access the checkbox
+    def is_checked(self):
+        return self._checkbox.isChecked()
+
+    def set_checked(self, value):
+        return self._checkbox.setChecked(value)
+
+    def set_check_visible(self, value):
+        self._checkbox.setVisible(value)
+
+    def is_check_visible(self):
+        self._checkbox.isVisible()
+
+    def checkbox_text(self):
+        self._checkbox.text()
+
+    def set_checkbox_text(self, text):
+        self._checkbox.setText(text)
+
+
+class WorkerUpdates(QObject):
+    """
+    Worker that checks for releases using the Github API without blocking the
+    Spyder user interface, in case of connections issues.
+    """
+    sig_ready = Signal()
+
+    def __init__(self, parent, feedback):
+        QObject.__init__(self)
+        self._parent = parent
+        self.feedback = feedback
+        self.error = feedback
+        self.latest_release = None
+
+    def is_stable_version(self, version):
+        """
+        A stable version has no letters in the final part, it has only numbers.
+
+        Stable version example: 1.2, 1.3.4, 1.0.5
+        Not stable version: 1.2alpha, 1.3.4beta, 0.1.0rc1, 3.0.0dev
+        """
+        if not isinstance(version, tuple):
+            version = version.split('.')
+        last_part = version[-1]
+
+        try:
+            int(last_part)
+            return True
+        except ValueError:
+            return False
+
+    def check_update_available(self, version, releases):
+        """Checks if there is an update available.
+
+        It takes as parameters the current version of Spyder and a list of
+        valid cleaned releases in chronological order (what github api returns
+        by default). Example: ['2.3.4', '2.3.3' ...]"""
+        if self.is_stable_version(version):
+            # Remove non stable versions from the list
+            releases = [r for r in releases if self.is_stable_version(r)]
+
+        latest_release = releases[0]
+
+        if version.endswith('dev'):
+            return (False, latest_release)
+
+        # check_version is based on LooseVersion, so a small hack is needed so
+        # that LooseVersion understands that '3.0.0' is in fact bigger than
+        # '3.0.0rc1'
```

Comment:
One last thing: please move this hack to `check_version`. It'll be more useful there :-)